src/lib/storage/constraints/unique_checker.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <boost/container/small_vector.hpp>
+#include <functional>
+#include <memory>
+#include <set>
+#include <unordered_set>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "operators/validate.hpp"
+#include "resolve_type.hpp"
+#include "storage/constraints/table_constraint_definition.hpp"
+#include "storage/mvcc_data.hpp"
+#include "storage/segment_accessor.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+bool constraint_valid_for(const Table& table, const TableConstraintDefinition& constraint,
+                          const CommitID& snapshot_commit_id, const TransactionID& our_tid);
+bool all_constraints_valid_for(std::shared_ptr<const Table> table, const CommitID& snapshot_commit_id,
```

Comment:
Consistency: Why do you sometimes pass around `shared_ptr<Table>`, and sometimes `Table&`?

Reply:
Originally we started with `shared_ptr<Table>`, but `constraint_valid_for` needs to be called from inside `table.add_unique_constraint` where we don't have a shared pointer. Should we use const reference everywhere?